tools/codegen/codespec/merge_attributes.go

```diff
@@ -35,88 +28,105 @@ func mergeNestedAttributes(existingAttrs *Attributes, newAttrs Attributes, compu
-		} else if attr.MapNested != nil {
-			existingAttr.MapNested = attr.MapNested
-		}
-	} else {
-		// add new attribute with the given computability
-		newAttr := *attr
-		newAttr.ComputedOptionalRequired = computability
-		newAttr.ReqBodyUsage = reqBodyUsage
-		merged[attr.Name.SnakeCase()] = &newAttr
+	if existingAttr.MapNested != nil && newAttr.MapNested != nil {
+		mergeNestedAttributes(&existingAttr.MapNested.NestedObject.Attributes, newAttr.MapNested.NestedObject.Attributes, reqBodyUsage, isFromResponse)
+	} else if newAttr.MapNested != nil {
+		existingAttr.MapNested = newAttr.MapNested
 	}
 }
 
+type attributeDefinitionSources struct {
+	createPathParams, createRequest, updateRequest, createResponse, readResponse Attributes
+}
+
 // mergeAttributes merges attributes from different sources (path params, create/get operation bodies) and determines a single merged list of attributes.
 // Computability and reqBodyUsage values are determined as part of this process.
 // Different sources are applied in a specific order, defining the computability and reqBodyUsage value they have at each step.
-func mergeAttributes(createPathParams, createRequest, createResponse, readResponse Attributes) Attributes {
+func mergeAttributes(sources *attributeDefinitionSources) Attributes {
 	merged := make(map[string]*Attribute)
 
 	// create path parameters: all attributes will be "required", reqBodyUsage is defined as omit all at this step
-	for i := range createPathParams {
-		addOrUpdate(&createPathParams[i], Required, OmitAll, merged, false)
+	for i := range sources.createPathParams {
+		addOrUpdate(merged, &sources.createPathParams[i], OmitAll, false)
 	}
 
-	// POST request body: optional/required is as defined, reqBodyUsage is defined as AllRequestBodies
-	for i := range createRequest {
+	// POST request body: optional/required is as defined, reqBodyUsage is defined as OmitUpdateBody and will be updated to AllRequestBodies if present in POST request
+	for i := range sources.createRequest {
 		// for now we do not differentiate AllRequestBodies vs PostBodyOnly as we are not processing update request
-		addOrUpdate(&createRequest[i], createRequest[i].ComputedOptionalRequired, AllRequestBodies, merged, false)
+		addOrUpdate(merged, &sources.createRequest[i], OmitUpdateBody, false)
+	}
+
+	// PATCH request body: optional/required is as defined, reqBodyUsage is defined as AllRequestBodies
+	for i := range sources.updateRequest {
+		addOrUpdate(merged, &sources.updateRequest[i], AllRequestBodies, false)
 	}
 
 	// POST/GET response body: properties not in the request body are "computed" or "computed_optional" (if a default is present), reqBodyUsage will have OmitAll not present in request body
-	for i := range createResponse {
-		if hasDefault(&createResponse[i]) {
-			addOrUpdate(&createResponse[i], ComputedOptional, OmitAll, merged, true)
-		} else {
-			addOrUpdate(&createResponse[i], Computed, OmitAll, merged, true)
-		}
```

Comment:
when properties are parsed in request/responses of create/update computability is defined as Optional or ComputedOptional depending if a default value is present, this logic was redundant